### PR TITLE
Add compiler-team repository under automation

### PR DESCRIPTION
**Pull request by Kobzol**
Repo: https://github.com/rust-lang/compiler-team

Current permissions extracted from GH:
```
org = "rust-lang"
name = "compiler-team"
description = "A home for compiler team planning documents, meeting minutes, and other such things."
bots = []

[access.teams]
wg-prioritization = "write"
compiler-contributors = "write"
compiler = "admin"
security = "pull"
mods = "write"
types = "write"
wg-compiler-performance = "write"

[access.individuals]
Nadrieril = "write"
lqd = "write"
oli-obk = "admin"
aliemjay = "write"
nikic = "write"
jdno = "admin"
Aaron1011 = "admin"
rustbot = "write"
chenyukang = "write"
camelid = "write"
wesleywiser = "admin"
flodiebold = "write"
hkmatsumoto = "write"
rust-lang-owner = "admin"
technetos = "write"
petrochenkov = "admin"
nagisa = "admin"
RalfJung = "write"
est31 = "write"
fee1-dead = "write"
compiler-errors = "admin"
cjgillot = "admin"
jackh726 = "admin"
davidtwco = "admin"
eholk = "write"
BoxyUwU = "write"
tgnottingham = "write"
fmease = "write"
eddyb = "admin"
cuviper = "write"
b-naber = "write"
nnethercote = "write"
hameerabbasi = "write"
estebank = "admin"
Nilstrieb = "write"
matthewjasper = "admin"
durin42 = "write"
badboy = "admin"
the8472 = "write"
saethlin = "write"
tmandry = "write"
matklad = "write"
inquisitivecrystal = "write"
TaKO8Ki = "write"
pnkfelix = "admin"
bjorn3 = "write"
michaelwoerister = "admin"
SparrowLii = "write"
tmiasko = "write"
Dylan-DPC = "write"
lcnr = "admin"
Mark-Simulacrum = "admin"
pietroalbini = "admin"
rylev = "admin"
apiraino = "write"
nikomatsakis = "write"
Kobzol = "write"
WaffleLapkin = "write"
spastorino = "write"
rfcbot = "write"
```

I kept only the teams that made sense to me.

### CODE REVIEW

**Comment by Kobzol** [comment]
Added `required-approvals = 0` to allow insta-merging of PRs.

File: repos/rust-lang/compiler-team.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+org = "rust-lang"
+name = "compiler-team"
+description = "Contains a static site with procedures, policies, working groups, planning documents and minutes of the compiler team."
+bots = []
+
+[access.teams]
+compiler = "write"
```
Comment (compiler-errors) [inline repos/rust-lang/compiler-team.toml:7]: please give compiler contributors access to the repo

Comment (Kobzol) [reply repos/rust-lang/compiler-team.toml:7]: Done!

File: repos/rust-lang/compiler-team.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+org = "rust-lang"
+name = "compiler-team"
+description = "A home for compiler team planning documents, meeting minutes, and other such things."
+bots = []
```
Comment (rylev) [inline repos/rust-lang/compiler-team.toml:4]: Does rustbot need to be added here?

Comment (Kobzol) [reply repos/rust-lang/compiler-team.toml:4]: Is it used there? Just to make sure I understand it properly, does the existence of `triagebot.toml` in the root of the repo imply that the repo used rustbot?

Comment (rylev) [reply repos/rust-lang/compiler-team.toml:4]: It is used (e.g., [here](https://github.com/rust-lang/compiler-team/issues/710#issuecomment-1875578907)). I believe the existence of that config does imply that it's used, but I was just going by issues where rustbot made an appearance.

Comment (Kobzol) [reply repos/rust-lang/compiler-team.toml]: Ok, added it.

**Review by rylev — APPROVED** [review]

**Pull request merged**